display.go

```diff
@@ -12,16 +12,79 @@ import (
 	"fyne.io/fyne/v2/canvas"
 	"fyne.io/fyne/v2/container"
 	"fyne.io/fyne/v2/driver/desktop"
+	"fyne.io/fyne/v2/widget"
+	"tinygo.org/x/drivers/touch"
 )
 
+type touchWidget struct {
+	widget.BaseWidget
+	obj fyne.CanvasObject
+	d   *Device
+}
+
+func (s *touchWidget) CreateRenderer() fyne.WidgetRenderer {
+	return widget.NewSimpleRenderer(s.obj)
+}
+
+func (s *touchWidget) setTouchPoint(p fyne.Position) {
+
+	x := int(p.X)
+	y := int(p.Y)
+
+	//Clamp X & Y values to width and height. If the pointer is moved outside the window these values may be negative or larger than the window
+	if x < 0 {
+		x = 0
+	} else if x > s.d.Width {
+		x = s.d.Width - 1
+	}
+
+	if y < 0 {
+		y = 0
+	} else if y > s.d.Height {
+		y = s.d.Height - 1
+	}
```

Comment:
```suggestion
	if x < 0 {
		x = 0
	} else if x >= s.d.Width {
		x = s.d.Width - 1
	}

	if y < 0 {
		y = 0
	} else if y >= s.d.Height {
		y = s.d.Height - 1
	}
```

Reply:
I'ce made the change. Sorry I should have noticed that myself.